src/components/common/Dialogs/CreatePaymentDialog/TokenAmountInput/TokenAmountInput.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useMemo } from 'react';
+import { defineMessages } from 'react-intl';
+import { AddressZero } from '@ethersproject/constants';
+import { useFormContext } from 'react-hook-form';
+
+import { HookFormInput as Input, TokenSymbolSelector } from '~shared/Fields';
+import EthUsd from '~shared/EthUsd';
+import { getTokenDecimalsWithFallback } from '~utils/tokens';
+import { notNull } from '~utils/arrays';
+import { Colony } from '~types';
+
+// import NetworkFee from '../NetworkFee';
+
+import styles from './TokenAmountInput.css';
+
+const displayName =
+  'common.CreatePaymentDialog.CreatePaymentDialogForm.TokenAmountInput';
+
+const MSG = defineMessages({
+  amount: {
+    id: `${displayName}.amount`,
+    defaultMessage: 'Amount',
+  },
+  fee: {
+    id: `${displayName}.fee`,
+    defaultMessage: 'Network fee: {fee} {symbol}',
+  },
+  token: {
+    id: `${displayName}.address`,
+    defaultMessage: 'Token',
+  },
+});
+
+interface Props {
+  colony: Colony | undefined;
+  disabled: boolean;
+}
+
+const TokenAmountInput = ({ colony, disabled }: Props) => {
+  const { getValues } = useFormContext();
+  const values = getValues();
+  const colonyTokens =
+    colony?.tokens?.items
+      .filter(notNull)
+      .map((colonyToken) => colonyToken.token) || [];
+  const selectedToken = colonyTokens.find(
+    (token) => token?.tokenAddress === values.tokenAddress,
+  );
+  const formattingOptions = useMemo(
```

Comment:
I'm gonna have to cite one of the greatest thinker from our generation, Will from 2 months ago:

> * @NOTE: If formattingOptions is not either memoized or defined outside of the ancestor Input component,
> * it will cause Cleave to be re-mounted and thus lose its state and focus.


If I don't use `useMemo` for `formattingOptions`, everytime I type something I lose focus :(